Commit 56d7747a3cf811910c4cf865e1ebcb8b82502005 breaks stubdom PCI
passthrough where MSI-X is present.  MiniOS mmap() will abort() if it does
not get PROT_WRITE | PROT_READ.

Signed-off-by: Eric Shelton <eshelton@pobox.com>

--- a/tools/qemu-xen-traditional/hw/pt-msi.c	2016-11-01 08:38:03.004785559 -0400
+++ b/tools/qemu-xen-traditional/hw/pt-msi.c	2016-11-01 08:40:49.144785559 -0400
@@ -586,7 +586,11 @@
     PT_LOG("table_off = %x, total_entries = %d\n", table_off, total_entries);
     dev->msix->table_offset_adjust = table_off & 0x0fff;
     dev->msix->phys_iomem_base = mmap(0, total_entries * 16 + dev->msix->table_offset_adjust,
+#ifdef CONFIG_STUBDOM
+                          PROT_WRITE | PROT_READ, MAP_SHARED | MAP_LOCKED,
+#else
                           PROT_READ, MAP_SHARED | MAP_LOCKED,
+#endif
                           fd, dev->msix->table_base + table_off - dev->msix->table_offset_adjust);
     close(fd);
     if ( dev->msix->phys_iomem_base == MAP_FAILED )
